pycytominer/cyto_utils/DeepProfiler_processing.py

```diff
@@ -190,27 +200,19 @@ def aggregate_deep(self):
                     f"No files for the key {metadata_level} could be found.\nThis program will continue, but be aware that this might induce errors!"
                 )
                 continue
-
             df = pd.concat(arr)
 
-            # Prepare inputs for the aggregate function
-            meta_df = pd.DataFrame(
-                self.file_aggregate[metadata_level]["metadata"], index=[0]
-            ).reset_index(drop=True)
-            meta_df.columns = [
-                f"Metadata_{x.capitalize()}" if not x.startswith("Metadata_") else x
-                for x in meta_df.columns
-            ]
+            # extract metadata prior to aggregation
+            meta_df = pd.DataFrame()
+            metadata_cols = [x for x in df.columns if x.startswith("Metadata_")]
```

Comment:
You could call `infer_cp_features(metadata=True)` from `pycytominer.cyto_utils()`